.github/workflows/test.yml

```diff
@@ -0,0 +1,30 @@
+# A simple workflow to run tests using pytest against supported Python versions
+
+name: Tests
+on: [ pull_request ]
```

Comment:
[consider][minor] [allow/ignore](https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions#example-ignoring-paths) appropriate paths, so we don't run pytest if only `doc` changes. We could only "allow" the `python` directory, but that's perhaps less robust in case we add python files elsewhere, we'd have to update this rule.

Reply:
Ahh, good catch, updated [here](https://github.com/TheFoundryVisionmongers/OpenAssetIO/pull/91/commits/82f2b0d0c1e48eb55fa5050f319a19c48f3d6235) - dummy PR [here](https://github.com/foundrytom/OpenAssetIO/pull/4), showing the check is skipped for the `docs` branch.

Reply:
Good idea to do that too, though not quite what I meant - I was talking about skipping based on paths in the main repo.  Also the dummy PR you link still runs pytest :(

Reply:
Doh! Im sure I looked at that and it exited early! 🙃 For some reason, despite the PR base being `docs` it's insisting on merging into `main` 🤷:
```
HEAD is now at 96827ef Merge 82f2b0d0c1e48eb55fa5050f319a19c48f3d6235 into e8f82dd47e83d5e7a62d163d3ae7b63d3b239491
```

`e8f82` is `HEAD` on `main`....

Reply:
It _seems_ like something to do with the run definition changes when you push, not just re-run. I pushed [this](https://github.com/TheFoundryVisionmongers/OpenAssetIO/pull/91/commits/7e996aa0e1747b3cd6193cf4c45450ba876fadd8) commit, and it seems to have properly skipped the check [here](https://github.com/foundrytom/OpenAssetIO/pull/2), and they seem to be running [here](https://github.com/foundrytom/OpenAssetIO/pull/5).

Reply:
It's definitely mixing up runs. I made a fresh PR -> `docs` on my fork, and its auto-linking the check to the results from the old PR 😠